src/OpenSearch/Client.php

```diff
@@ -65,214 +68,110 @@ class Client
 {
     public const VERSION = '2.3.1';
 
-    /**
-     * @var Transport
-     */
-    public $transport;
+    protected bool $isAsync = false;
 
-    /**
-     * @var array
-     */
-    protected $params;
+    protected AsyncSearchNamespace $asyncSearch;
 
-    /**
-     * @var callable
-     */
-    protected $endpoints;
+    protected AsynchronousSearchNamespace $asynchronousSearch;
 
-    /**
-     * @var NamespaceBuilderInterface[]
-     */
-    protected $registeredNamespaces = [];
+    protected CatNamespace $cat;
 
-    /**
-     * @var AsyncSearchNamespace
-     */
-    protected $asyncSearch;
+    protected ClusterNamespace $cluster;
 
-    /**
-     * @var AsynchronousSearchNamespace
-     */
-    protected $asynchronousSearch;
+    protected DanglingIndicesNamespace $danglingIndices;
 
-    /**
-     * @var CatNamespace
-     */
-    protected $cat;
+    protected DataFrameTransformDeprecatedNamespace $dataFrameTransformDeprecated;
 
-    /**
-     * @var ClusterNamespace
-     */
-    protected $cluster;
-
-    /**
-     * @var DanglingIndicesNamespace
-     */
-    protected $danglingIndices;
+    protected FlowFrameworkNamespace $flowFramework;
 
-    /**
-     * @var DataFrameTransformDeprecatedNamespace
-     */
-    protected $dataFrameTransformDeprecated;
-
-    /**
-     * @var FlowFrameworkNamespace
-     */
-    protected $flowFramework;
+    protected IndicesNamespace $indices;
 
-    /**
-     * @var IndicesNamespace
-     */
-    protected $indices;
+    protected IngestNamespace $ingest;
 
-    /**
-     * @var IngestNamespace
-     */
-    protected $ingest;
+    protected InsightsNamespace $insights;
 
-    /**
-     * @var InsightsNamespace
-     */
-    protected $insights;
+    protected IsmNamespace $ism;
 
-    /**
-     * @var IsmNamespace
-     */
-    protected $ism;
+    protected KnnNamespace $knn;
 
-    /**
-     * @var KnnNamespace
-     */
-    protected $knn;
+    protected MlNamespace $ml;
 
-    /**
-     * @var MlNamespace
-     */
-    protected $ml;
+    protected MonitoringNamespace $monitoring;
```

Comment:
Turning protected properties into typed properties is technically a BC break for child classes if they redefine the properties.

Btw, if this is going to be released as a major version of the package, maybe private visibility should be used instead (removing the properties themselves from the API covered by semver as child classes won't have access to them anymore). After all, the child classes can read them using the public getter.